site/public/js/forum.js

```diff
@@ -798,12 +798,17 @@ function showEditPostForm(post_id, thread_id, shouldEditThread, render_markdown,
                 var thread_title = json.title;
                 var thread_lock_date =  json.lock_thread_date;
                 var thread_status = json.thread_status;
+                var expiration = json.expiration;
                 $("#title").prop('disabled', false);
                 $(".edit_thread").show();
                 $('#label_lock_thread').show();
                 $("#title").val(thread_title);
                 $("#thread_status").val(thread_status);
                 $('#lock_thread_date').val(thread_lock_date);
+                if (Date.parse(expiration) > new Date()) {
```

Comment:
Have you tested this with conflicting timezones? Since the timezone is stored in the DB it should be fine but just to be sure.

Reply:
I have tested with conflicting time zones it seems to work correctly.